Add Minitest

blowmage opened the pull request:
This PR converts all remaining RSpec specs to Minitest. Right now the tests are using the Spec DSL to ease the review easier. We can convert the new tests to the assertion style in a future PR.

quartzmo commented on test/google/gax/error_test.rb:
```diff
@@ -0,0 +1,239 @@
+require 'test_helper'
+require 'google/gax/errors'
+require 'google/protobuf/any_pb'
+require_relative '../../../spec/fixtures/fixture_pb'
```
Does this file use types from `fixture_pb`? I don't see any usages. If not, can we remove this line?

blowmage replied:
It does not. That was ported from the spec file. Removed.

quartzmo commented on test/google/gax/util_test.rb:
```diff
@@ -0,0 +1,200 @@
+require 'google/gax'
+require 'google/protobuf/any_pb'
+require 'google/protobuf/timestamp_pb'
+require_relative '../../../spec/fixtures/fixture_pb'
```
Can `fixtures/fixture_pb.rb` be copied to the `test` dir in this PR to prepare for deleting `spec` in a follow-up PR?

blowmage replied:
Yes, but I didn't want to make any changes to the spec directory in this PR. I planned on moving these files when RSpec is removed.

quartzmo replied:
Can fixtures/fixture_pb.rb be copied to the test dir in this PR to prepare for deleting spec in a follow-up PR? Copying will not change the spec dir.

blowmage replied:
Eh, changed my mind, moved.

quartzmo commented on test/google/gax/util_test.rb:
```diff
@@ -0,0 +1,200 @@
+    end
+
+    it 'handles IO instances' do
+      file = File.new('spec/fixtures/fixture_file.txt')
```
Can `spec/fixtures/fixture_file.txt` be copied to the test dir in this PR to prepare for deleting spec in a follow-up PR?

blowmage replied:
Moved.

blowmage commented:
Pull and try again. I added a commit that added a `get_application_default` stub that I had removed earlier and I didn't think it was being used/needed.

quartzmo commented:
Credentials error fixed, thanks! 👍
